client/src/components/History/CurrentCollection/CollectionNavigation.vue

```diff
@@ -1,46 +1,58 @@
+<script setup lang="ts">
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faAngleDoubleLeft, faAngleLeft } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
+import { BButton } from "bootstrap-vue";
+import { computed } from "vue";
+
+library.add(faAngleDoubleLeft, faAngleLeft);
+
+interface Props {
+    historyName: string;
+    selectedCollections: object[];
```

Comment:
Try to use the proper interface here. It should also avoid the dubious casting down below on line 23.

Reply:
```suggestion
    selectedCollections: CollectionEntry[];
```
You can import it from:
```ts
import type { CollectionEntry } from "@/api";
```

Reply:
`element_identifier` does not exist in this type.
![image](https://github.com/galaxyproject/galaxy/assets/8046843/cf1b4a1b-336a-4e4a-8b8e-667033783bc8)
I can't find a type for it

Reply:
If you trace the type back to the component passing the props you can see that `CollectionEntry[]` is the right type for `selectedCollections`:

https://github.com/itisAliRH/galaxy/blob/368da66955868bff107ddb69edd20795084c8b80/client/src/components/History/CurrentCollection/CollectionPanel.vue#L114

What we need to fix is the condition that is showing the error. CollectionEntry can be of 2 different types (HDCASummary | SubCollection). Both HDCASummary and SubCollection should have a `name` and you can remove `|| last?.element_identifier` from the condition. The `element_identifier` is mapped to the `name` in the case of the SubCollection:

https://github.com/itisAliRH/galaxy/blob/368da66955868bff107ddb69edd20795084c8b80/client/src/components/History/CurrentCollection/CollectionPanel.vue#L83

The best thing about TS is that it helps identify code errors we didn't notice in JS, like in this condition. In general, we should "trust" the types that come from the API although there could be errors in those API types too. So using the proper type instead of creating an "ad-hoc type" can help us identify errors in both directions :tada:

Reply:
Thank you for pointing it out. I appreciate your detailed review!